### ISSUE TITLE
segfault when saving reference image

### ISSUE THREAD

**Issue by agriggio** [Beep6581/RawTherapee#4289]
As the title says. The problem seems to be in the TIFF writing when `exifRoot` is NULL. This patch fixes itt, but I'd like @Hombre57  to take a look...

```diff
diff --git a/rtengine/imageio.cc b/rtengine/imageio.cc
--- a/rtengine/imageio.cc
+++ b/rtengine/imageio.cc
@@ -1485,9 +1485,9 @@
     }
 
 #if __BYTE_ORDER__==__ORDER_LITTLE_ENDIAN__
-    bool needsReverse = (bps == 16 || bps == 32) && exifRoot->getOrder() == rtexif::MOTOROLA;
+    bool needsReverse = (bps == 16 || bps == 32) && exifRoot && exifRoot->getOrder() == rtexif::MOTOROLA;
 #else
-    bool needsReverse = (bps == 16 || bps == 32) && exifRoot->getOrder() == rtexif::INTEL;
+    bool needsReverse = (bps == 16 || bps == 32) && exifRoot && exifRoot->getOrder() == rtexif::INTEL;
 #endif
 
     for (int row = 0; row < height; row++) {
```

**Comment by Floessie** [comment]
@agriggio @Hombre57 This was also reported by yesterday's Coverity scan. I would have implemented exactly the same fix. 👍

**Comment by Beep6581** [comment]
Does this mean that when using "Save reference image for profiling", metadata is now included? That would be great.

**Comment by agriggio** [comment]
@Beep6581 nope, that's exactly the problem -- metadata is assumed to be there, but it is not...
I haven't changed that

**Comment by Hombre57** [comment]
@agriggio Looks like I overlooked this occurrence. Could you commit ?

@Beep6581 Looking at the missing metadata...

**Comment by Hombre57** [comment]
@agriggio 
> nope, that's exactly the problem -- metadata is assumed to be there, but it is not...

I guess that it's possible to have input file without metadata, so this test has to be there anyway.

**Comment by agriggio** [comment]
> Looks like I overlooked this occurrence. Could you commit ?

@Hombre57 done.

**Comment by Hombre57** [comment]
@Beep6581 I don't find any issue for the metadata problem. Do you know if one exist ?

**Comment by Beep6581** [comment]
@Hombre57 https://github.com/Beep6581/RawTherapee/issues/3567

**Comment by Hombre57** [comment]
@Beep6581 The metadata problem in the ICC reference file is fixed now (commit 91e494a7098e760d2422003011e8311d7032bba5).
